sympy/solvers/tests/test_solvers.py

```diff
@@ -1775,20 +1776,16 @@ def test_issue_8828():
 
 
 def test_issue_2840_8155():
-    assert solve(sin(3*x) + sin(6*x)) == [
-        0, -pi, pi, 2*pi, -2*I*log(-(-1)**(S(1)/9)), -2*I*log(-(-1)**(S(2)/9)),
-        -2*I*log((-1)**(S(7)/9)), -2*I*log((-1)**(S(8)/9)), -2*I*log(-S(1)/2 -
-        sqrt(3)*I/2), -2*I*log(-S(1)/2 + sqrt(3)*I/2), -2*I*log(S(1)/2 -
-        sqrt(3)*I/2), -2*I*log(S(1)/2 + sqrt(3)*I/2), -2*I*log(-sqrt(3)/2 - I/2),
-        -2*I*log(-sqrt(3)/2 + I/2), -2*I*log(sqrt(3)/2 - I/2),
```

Comment:
Yes , I'll look into making the evaluation correct instead and finding out where the spurious imaginary parts are produced.Currently my end semester exams are going on. I'll work on this again from the 15th.

It's seems like a problem with evaluation of functions (such as `log`, `exp`, etc) which can potentially yield imaginary terms and not with the algorithm in `solvers`.